r/tests/testthat/helper-expectation.R

```diff
@@ -22,3 +22,16 @@ expect_vector <- function(x, y, ...) {
 expect_data_frame <- function(x, y, ...) {
   expect_equal(as.data.frame(x), y, ...)
 }
+
+expect_equivalent <- function(object, expected, ...) {
+  # HACK: dplyr includes an all.equal.tbl_df method that is causing failures.
+  # They look spurious, like:
+  # `Can't join on 'b' x 'b' because of incompatible types (tbl_df/tbl/data.frame / tbl_df/tbl/data.frame)`
```

Comment:
@romainfrancois FYI. I see this noted on https://github.com/tidyverse/dplyr/issues/2751#issuecomment-460100759 (and probably some other issues, there seem to be quite a few relating to `all.equal.tbl_df`)